@types/index.d.ts

```diff
@@ -420,7 +413,14 @@ declare module "opentok-react-native" {
   /**
    * https://github.com/opentok/opentok-react-native/blob/master/docs/OTPublisher.md
    */
-  export class OTPublisher extends React.Component<OTPublisherProps> {}
+  export class OTPublisher extends React.Component<OTPublisherProps, unknown> {
```

Comment:
React.Component **requires** two type arguments: <Props, State>

I did not want to dive too deep in what should be declared for State; thus `unknown` is a good start.